src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java

```diff
@@ -147,6 +147,10 @@ public static void createBucketWithDataTypeTest2File() throws IOException, URISy
   public static void createBucketWithRecursiveTestFiles() throws IOException, URISyntaxException {
     gcsSourceBucketName = createGCSBucketWithMultipleFiles(PluginPropertyUtils.pluginProp("gcsReadRecursivePath"));
   }
+  @Before(order = 1, value = "@GCS_MULTIPLE_FILES_TEST")
+  public static void createBucketWithMultipleTestFiles() throws IOException, URISyntaxException {
+    gcsSourceBucketName = createGCSBucketWithMultipleFiles(PluginPropertyUtils.pluginProp("gcsMultipleFilesPath"));
+  }
```

Comment:
Just asking : If we can use the same hooks in regex also ?

Reply:
In the case for multiple files with different schema. We have different columns name and count in both the file as per mentioned in the ITN Class. On the other hand, multiple files with filter regex, we need to use the same schema(same columns) in those two files that we need to read. For example, author-part-1 and author-part-2 should have "Name;Surname;Age". So need to create new test data and hook for further validations.

Reply:
Ok got it , thanks